lib/auth/Vault.js

```diff
@@ -222,6 +222,40 @@ class Vault {
             });
     }
 
+    /** getAccountIds -- call Vault to get accountIds based on
+     * canonicalIDs
+     * @param {array} canonicalIDs - list of canonicalIDs
+     * @param {object} log - log object
+     * @param {function} callback - callback with either error or an object
+     * with canonicalID keys and accountId values
+     * @return {undefined}
+    */
+    getAccountIds(canonicalIDs, log, callback) {
+        log.trace('getting accountIds from Vault based on canonicalIDs',
+        { canonicalIDs });
+        this.client.getAccountIds(canonicalIDs,
+        { reqUid: log.getSerializedUids() },
+        (err, info) => {
+            if (err) {
+                log.debug('received error message from vault',
+                    { errorMessage: err });
+                return callback(err);
+            }
+            const infoFromVault = info.message.body;
+            log.trace('info received from vault', { infoFromVault });
+            const result = {};
+            /* If the accountId was not found in Vault, do not
+            send the canonicalID back to the API */
+            Object.keys(infoFromVault).forEach(key => {
+                if (infoFromVault[key] !== 'NotFound' &&
```

Comment:
Do we need to hide those return codes? Thinking they might be used by the caller for error management.

Reply:
Stripping the `NotFound` keys is consistent with the `getEmailAddresses` and `getCanonicalIds` methods.